src/emu/console/src/thread.cpp

```diff
@@ -45,11 +45,41 @@ void set_mouse_down(void *userdata, const int button, const bool op) {
     emu->mouse_down[button] = op;
 }
 
+static eka2l1::drivers::input_event make_mouse_event_driver(float x, float y, int button, int action) {
+    eka2l1::drivers::input_event evt = *new eka2l1::drivers::input_event();
+    evt.type_ = eka2l1::drivers::input_event_type::touch;
+    evt.mouse_.pos_x_ = static_cast<int>(x);
+    evt.mouse_.pos_y_ = static_cast<int>(y);
+    evt.mouse_.button_ = button == 0 ? eka2l1::drivers::mouse_button::left :
+                        (button == 1 ? eka2l1::drivers::mouse_button::right :
+                        eka2l1::drivers::mouse_button::middle);
+    evt.mouse_.action_ = action == 0 ? eka2l1::drivers::mouse_action::press :
+                        (action == 1 ? eka2l1::drivers::mouse_action::repeat :
+                        eka2l1::drivers::mouse_action::release);
+    return evt;
+}
+
+/**
+ * \brief Callback when a host mouse event is triggered
+ * \param mouse_pos    position of mouse pointer          
+ * \param button       0: left, 1: right, 2: other      
+ * \param action       0: press, 1: repeat(move), 2: release      
+ */
 static void on_ui_window_mouse_evt(void *userdata, eka2l1::point mouse_pos, int button, int action) {
+    float mouse_pos_x = static_cast<float>(mouse_pos.x), mouse_pos_y = static_cast<float>(mouse_pos.y);
+
     eka2l1::desktop::emulator *emu = reinterpret_cast<eka2l1::desktop::emulator *>(userdata);
-    float scale = emu->symsys->get_config()->ui_scale;
+    if (emu->symsys) {
+        float scale = emu->symsys->get_config()->ui_scale;
+        mouse_pos_x /= scale;
+        mouse_pos_y /= scale;
+
+        auto mouse_evt = make_mouse_event_driver(mouse_pos_x, mouse_pos_y, button, action);
```

Comment:
const

Reply:
mouse_evt was passed to queue_input_from_driver next line which doesn't accept const parameter